rclcpp/src/rclcpp/node_interfaces/node_topics.cpp

```diff
@@ -51,9 +51,14 @@ NodeTopics::create_publisher(
       publisher_factory.add_publisher_to_intra_process_manager(ipm.get(), publisher);
     // Create a function to be called when publisher to do the intra process publish.
     auto shared_publish_callback = publisher_factory.create_shared_publish_callback(ipm);
+    auto count_callback = std::bind(
+      &rclcpp::intra_process_manager::IntraProcessManager::get_subscription_count,
+      ipm.get(),
+      std::placeholders::_1);
```

Comment:
@ivanpauno meta: would it make for the publisher to keep a reference to the IPM instead of using a bound method like this?